model/metrics/dotnet-metrics.yaml

```
@@ -0,0 +1,190 @@
+      garbage collection.
+    note: "This metric reports the same values as calling [`GC.GetGCMemoryInfo().GenerationInfo.SizeAfterBytes`](https://learn.microsoft.com/dotnet/api/system.gcgenerationinfo.sizeafterbytes)."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: dotnet.gc.heap.generation
+        requirement_level: required
+
+  - id: metric.dotnet.gc.heap.fragmentation
+    type: metric
+    metric_name: dotnet.gc.heap.fragmentation
+    brief: >
+      The heap fragmentation, as observed during the latest garbage collection.
+    note: >
+      This metric reports the same values as calling
+      [`GC.GetGCMemoryInfo().GenerationInfo.FragmentationAfterBytes`](https://learn.microsoft.com/dotnet/api/system.gcgenerationinfo.fragmentationafterbytes).
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+    attributes:
+      - ref: dotnet.gc.heap.generation
+        requirement_level: required
+
+  - id: metric.dotnet.gc.pause.time
+    type: metric
+    metric_name: dotnet.gc.pause.time
+    brief: "The total amount of time paused in GC since the process has started."
+    note: "This metric reports the same values as calling [`GC.GetTotalPauseDuration()`](https://learn.microsoft.com/dotnet/api/system.gc.gettotalpauseduration)."
+    instrument: counter
+    unit: "s"
+    stability: experimental
+
+  - id: metric.dotnet.jit.compiled_il.size
+    type: metric
+    metric_name: dotnet.jit.compiled_il.size
+    brief: "Count of bytes of intermediate language that have been compiled since the process has started."
+    note: "This metric reports the same values as calling [`JitInfo.GetCompiledILBytes()`](https://learn.microsoft.com/dotnet/api/system.runtime.jitinfo.getcompiledilbytes)."
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.dotnet.jit.compiled_method.count
+    type: metric
+    metric_name: dotnet.jit.compiled_method.count
+    brief: >
+      The number of times the JIT compiler (re)compiled methods since the
+      process has started.
+    note: "This metric reports the same values as calling [`JitInfo.GetCompiledMethodCount()`](https://learn.microsoft.com/dotnet/api/system.runtime.jitinfo.getcompiledmethodcount)."
+    instrument: counter
+    unit: "{method}"
+    stability: experimental
+
+  - id: metric.dotnet.jit.compilation.time
+    type: metric
+    metric_name: dotnet.jit.compilation.time
+    brief: >
+      The amount of time the JIT compiler has spent compiling methods since the
+      process has started.
+    note: "This metric reports the same values as calling [`JitInfo.GetCompilationTime()`](https://learn.microsoft.com/dotnet/api/system.runtime.jitinfo.getcompilationtime)."
+    instrument: counter
+    unit: "s"
+    stability: experimental
+
+  - id: metric.dotnet.monitor.lock_contention.count
+    type: metric
+    metric_name: dotnet.monitor.lock_contention.count
```

since it's a counter, 

```suggestion
    metric_name: dotnet.monitor.lock_contentions
```